[Test] Oomph Setup QA

The following scenarios should be tested to validate the Oomph based setup:

- [ ] Install a new developer's workspace on OSX and verify that all projects are present in the workspace, there are no errors in the workspace, and the emulator starts without error
- [ ] Repeat above for Linux

tested the Oomph installer to setup a developers workspace on a Windows 7 machine (shown at https://www.youtube.com/watch?v=RwE3tLtejHM&feature=youtu.be). Last year in August I was able (after some problems) to setup the development environment on windows 7. Therefore I used now again this approach and faced now following problems:
**Problem 1** occurred when the installer wanted to download the source code of kura from git (error logs say that the authentication failed: java.lang.Exception: org.eclipse.jgit.api.errors.TransportException: git@github.com:eclipse/kura.git: Auth fail). I manually cloned the code to the path (no authentication was used) and then let Oomph the steps afterwards. Had this problem already last year, and solved it by cloning the github repo manually
**Problem 2** Plugin org.eclipse.tycho:tycho-maven-plugin:0.26.0 cannot be resolved:
[INFO] Scanning for projects...
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/eclipse/tycho/tycho-maven-plugin/0.26.0/tycho-maven-plugin-0.26.0.pom
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] Unresolveable build extension: Plugin org.eclipse.tycho:tycho-maven-plugin:0.26.0 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.eclipse.tycho:tycho-maven-plugin:jar:0.26.0 @ 
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project org.eclipse.kura:kura:3.0.0-SNAPSHOT (C:\IoT\EclipseKura\EclipseInstaller\iot-kura-develop-2017\git\kura\kura\manifest_pom.xml) has 1 error
[ERROR]     Unresolveable build extension: Plugin org.eclipse.tycho:tycho-maven-plugin:0.26.0 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.eclipse.tycho:tycho-maven-plugin:jar:0.26.0: Could not transfer artifact org.eclipse.tycho:tycho-maven-plugin:pom:0.26.0 from/to central (https://repo.maven.apache.org/maven2): java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty -> [Help 2]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
[ERROR] [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/PluginManagerException
Had this problem already last year, and solved it last year. Tried to remember what I did last year and searched on the internet, but could not solve yet the problem.
**Problem 3**: compile errors in workspace -> target definition(kura\target-definition\kura-equinox_3.11.1.target) points to some directions that do no longer exist (f. ex. kura\target-definition\common\repository\plugins, kura\target-definition\equinox_3.8.1\repository\plugins). "Set target platform" like described here https://eclipse.github.io/kura/doc/kura-setup.html does not work.

Does somebody know how to solve the Tycho problem and the no longer valid paths in the target definition?
Thanks in advance

@dwoodard1: thank you for your comment. I changed in the Eclipse Installer, "Variables" view to "HTTP read / write", entered my mail address. Before clicking at the button "Finish" I added in Eclipse (Neon 2) following settings under "Window" -> "Preferences":
"General" -> "Network Connection" -> "SSH2": double check here if private ssh key is set (how this can be done is described here: https://help.github.com/enterprise/2.2/user/articles/generating-an-ssh-key/)
"Team" -> "Git" -> "Configuration": http.sslVerify=false; http.postBuffer=52428800
Then clicking at "Finish" in the Eclipse Installer, code was loaded successfully. 

At the step of launch setups some warnings happen:
Performing Launch setups/launchers/build-target-platform.launch
Execute Debug Command
Connecting Git team provider.
Updating indexes
[INFO] Scanning for projects... 

[WARNING]  
[WARNING] Some problems were encountered while building the effective model for org.eclipse.kura:org.usb4java:bundle:1.2.1-SNAPSHOT 

[WARNING] 'dependencies.dependency.systemPath' for org.usb4java:org.usb4java:jar should not point at files within the project directory, ${basedir}/src/lib/usb4java-1.2.0.jar will be unresolvable by dependent projects @ line 42, column 25 
[WARNING] 'dependencies.dependency.systemPath' for org.usb4java:org.usb4java.linux-arm:jar should not point at files within the project directory, ${basedir}/src/lib/libusb4java-1.2.0-linux-arm.jar will be unresolvable by dependent projects @ line 50, column 25 
[WARNING] 'dependencies.dependency.systemPath' for org.usb4java:org.usb4java.linux-x86_64:jar should not point at files within the project directory, ${basedir}/src/lib/libusb4java-1.2.0-linux-x86_64.jar will be unresolvable by dependent projects @ line 58, column 25 
[WARNING] 'dependencies.dependency.systemPath' for org.usb4java:org.usb4java.linux-x86:jar should not point at files within the project directory, ${basedir}/src/lib/libusb4java-1.2.0-linux-x86.jar will be unresolvable by dependent projects @ line 66, column 25 
[WARNING] 'dependencies.dependency.systemPath' for org.usb4java:org.usb4java.osx-x86:jar should not point at files within the project directory, ${basedir}/src/lib/libusb4java-1.2.0-osx-x86.jar will be unresolvable by dependent projects @ line 74, column 25 
[WARNING] 'dependencies.dependency.systemPath' for org.usb4java:org.usb4java.osx-x86_64:jar should not point at files within the project directory, ${basedir}/src/lib/libusb4java-1.2.0-osx-x86_64.jar will be unresolvable by dependent projects @ line 82, column 25 
[WARNING] 'dependencies.dependency.systemPath' for org.usb4java:org.usb4java.windows-x86_64:jar should not point at files within the project directory, ${basedir}/src/lib/libusb4java-1.2.0-windows-x86_64.jar will be unresolvable by dependent projects @ line 91, column 25 
[WARNING] 'dependencies.dependency.systemPath' for org.usb4java:org.usb4java.windows-x86:jar should not point at files within the project directory, ${basedir}/src/lib/libusb4java-1.2.0-windows-x86.jar will be unresolvable by dependent projects @ line 99, column 25 
[WARNING]  
[WARNING] Some problems were encountered while building the effective model for org.eclipse.kura:usb4java-javax:bundle:1.2.1-SNAPSHOT 
[WARNING] 'dependencies.dependency.systemPath' for javax.usb:usb-api:jar should not point at files within the project directory, ${basedir}/src/lib/usb-api-1.0.2.jar will be unresolvable by dependent projects @ line 32, column 25 
[WARNING] 'dependencies.dependency.systemPath' for usb4java-javax:org.usb4java.javax:jar should not point at files within the project directory, ${basedir}/src/lib/usb4java-javax-1.2.0.jar will be unresolvable by dependent projects @ line 40, column 25 
[WARNING]  
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build. 
[WARNING]  
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects. 
[WARNING]  

These warnings are for usb part, which results then in build failure:
[INFO] ------------------------------------------------------------------------ 
[INFO] Reactor Summary: 
[INFO]  
[INFO] target-platform .................................... SUCCESS [  1.209 s] 
[INFO] Camel :: Core :: Import sun.misc ................... SUCCESS [  1.672 s] 
[INFO] System extension fragment for 'sun.misc' ........... SUCCESS [  0.222 s] 
[INFO] Java API for working with Human Interface USB Devices (HID) FAILURE [  0.618 s] 
[INFO] usb4java ........................................... SKIPPED 
[INFO] usb4java-javax ..................................... SKIPPED 
[INFO] Serial Device based on SODA DK comm ................ SKIPPED 
[INFO] jdk.dio-parent ..................................... SKIPPED 
[INFO] jdk.dio ............................................ SKIPPED 
[INFO] jdk.dio.armv5sf .................................... SKIPPED 
[INFO] jdk.dio.armv6hf .................................... SKIPPED 
[INFO] jdk.dio.x86 ........................................ SKIPPED 
[INFO] jdk.dio.x86_64 ..................................... SKIPPED 
[INFO] gwt-user ........................................... SKIPPED 
[INFO] moquette-broker .................................... SKIPPED 
[INFO] p2-repo-common ..................................... SKIPPED 
[INFO] p2-repo-equinox_3.11.1 ............................. SKIPPED 
[INFO] p2-repo-tests-deps ................................. SKIPPED 
[INFO] ------------------------------------------------------------------------ 
[INFO] BUILD FAILURE 
[INFO] ------------------------------------------------------------------------ 
[INFO] Total time: 5.579 s 
[INFO] Finished at: 2017-01-25T15:43:10+01:00 

[INFO] Final Memory: 20M/104M 
[INFO] ------------------------------------------------------------------------ 

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:2.17:check (checkstyle-validation) on project com.codeminders.hidapi: Failed during checkstyle execution: There are 7 errors reported by Checkstyle 6.11.2 with ../checkstyle_checks.xml ruleset. -> [Help 1] 
[ERROR]  
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch. 
[ERROR] Re-run Maven using the -X switch to enable full debug logging. 
[ERROR]  
[ERROR] For more information about the errors and possible solutions, please read the following articles: 
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException 
[ERROR]  
[ERROR] After correcting the problems, you can resume the build with the command 
[ERROR]   mvn <goals> -rf :com.codeminders.hidapi 

Finally following error happens:
Updating Git status for repository kura:  12% ( 390/3014)
[WARNING] No explicit target runtime environment configuration. Build is platform dependent. 

[INFO] Computing target platform for MavenProject: org.eclipse.kura:org.eclipse.kura.api:1.2.0-SNAPSHOT @ C:\IoT\EclipseKura\EclipseInstaller\iot-kura-develop-2017\git\kura\kura\org.eclipse.kura.api\pom.xml 

Searching for markers
Process resource updates
[INFO] Adding repository file://C:/IoT/EclipseKura/EclipseInstaller/iot-kura-develop-2017/git/kura/kura/org.eclipse.kura.api/../target-definition/equinox_3.11.1/repository 

[ERROR] Internal error: java.lang.RuntimeException: Failed to load p2 repository with ID 'p2-repo-equinox_3.11.1' from location file:////C:/IoT/EclipseKura/EclipseInstaller/iot-kura-develop-2017/git/kura/kura/org.eclipse.kura.api/../target-definition/equinox_3.11.1/repository/: URI has an authority component -> [Help 1] 

	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:121) 

	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863) 
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288) 
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199) 

	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) 
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) 
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) 
	at java.lang.reflect.Method.invoke(Unknown Source) 
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289) 
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229) 
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415) 

	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356) 

Caused by: java.lang.RuntimeException: Failed to load p2 repository with ID 'p2-repo-equinox_3.11.1' from location file:////C:/IoT/EclipseKura/EclipseInstaller/iot-kura-develop-2017/git/kura/kura/org.eclipse.kura.api/../target-definition/equinox_3.11.1/repository/ 
	at org.eclipse.tycho.p2.target.TargetPlatformFactoryImpl.loadMetadataRepository(TargetPlatformFactoryImpl.java:302) 

	at org.eclipse.tycho.p2.target.TargetPlatformFactoryImpl.gatherExternalInstallableUnits(TargetPlatformFactoryImpl.java:270) 
	at org.eclipse.tycho.p2.target.TargetPlatformFactoryImpl.createTargetPlatform(TargetPlatformFactoryImpl.java:176) 
	at org.eclipse.tycho.p2.target.TargetPlatformFactoryImpl.createTargetPlatform(TargetPlatformFactoryImpl.java:128) 
	at org.eclipse.tycho.p2.target.TargetPlatformFactoryImpl.createTargetPlatform(TargetPlatformFactoryImpl.java:1) 
	at org.eclipse.tycho.p2.manager.ReactorRepositoryManagerImpl.computePreliminaryTargetPlatform(ReactorRepositoryManagerImpl.java:84) 
	at org.eclipse.tycho.p2.resolver.P2DependencyResolver.computePreliminaryTargetPlatform(P2DependencyResolver.java:223) 
	at org.eclipse.tycho.core.resolver.DefaultTychoResolver.resolveProject(DefaultTychoResolver.java:109) 
	at org.eclipse.tycho.core.maven.TychoMavenLifecycleParticipant.afterProjectsRead(TychoMavenLifecycleParticipant.java:95) 
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:266) 
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193) 
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106) 

	... 11 more 
Caused by: org.eclipse.equinox.p2.core.ProvisionException: URI has an authority component 
	at org.eclipse.equinox.internal.p2.engine.ProfileMetadataRepository.<init>(ProfileMetadataRepository.java:50) 

	at org.eclipse.equinox.internal.p2.engine.ProfileMetadataRepositoryFactory.load(ProfileMetadataRepositoryFactory.java:36) 
	at org.eclipse.equinox.internal.p2.metadata.repository.MetadataRepositoryManager.factoryLoad(MetadataRepositoryManager.java:57) 
	at org.eclipse.equinox.internal.p2.repository.helpers.AbstractRepositoryManager.loadRepository(AbstractRepositoryManager.java:768) 
	at org.eclipse.equinox.internal.p2.repository.helpers.AbstractRepositoryManager.loadRepository(AbstractRepositoryManager.java:668) 
	at org.eclipse.equinox.internal.p2.metadata.repository.MetadataRepositoryManager.loadRepository(MetadataRepositoryManager.java:96) 

	at org.eclipse.tycho.p2.remote.RemoteMetadataRepositoryManager.loadRepository(RemoteMetadataRepositoryManager.java:61) 
	at org.eclipse.tycho.p2.remote.RemoteMetadataRepositoryManager.loadRepository(RemoteMetadataRepositoryManager.java:53) 
	at org.eclipse.tycho.p2.target.TargetPlatformFactoryImpl.loadMetadataRepository(TargetPlatformFactoryImpl.java:298) 
	... 22 more 

Caused by: java.lang.IllegalArgumentException: URI has an authority component 
	at java.io.File.<init>(Unknown Source) 

	at org.eclipse.equinox.internal.p2.engine.ProfileMetadataRepository.getProfile(ProfileMetadataRepository.java:173) 
	at org.eclipse.equinox.internal.p2.engine.ProfileMetadataRepository.<init>(ProfileMetadataRepository.java:48) 
	... 30 more 
[ERROR]  

[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch. 
[ERROR] Re-run Maven using the -X switch to enable full debug logging. 
[ERROR]  
[ERROR] For more information about the errors and possible solutions, please read the following articles: 

[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/InternalErrorException

executed _mvn compile_ in git\kura\target-platform path provided more details: seams that in some code files Windows line endings are used:
[INFO] ------------------------------------------------------------------------
[INFO] Building Java API for working with Human Interface USB Devices (HID) 1.1.3-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-checkstyle-plugin:2.17:check (checkstyle-validation) @ com.codeminders.hidapi ---
[INFO] Starting audit...
C:\IoT\EclipseKura\EclipseInstaller\iot-kura-develop-2017\git\kura\target-platform\com.codeminders.hidapi\src\main\java\com\codeminders\hidapi\ClassPathLibraryLoader.java:1: error: Do not use Windows line endings
C:\IoT\EclipseKura\EclipseInstaller\iot-kura-develop-2017\git\kura\target-platform\com.codeminders.hidapi\src\main\java\com\codeminders\hidapi\HIDAPITest.java:1: error: Do not use Windows  line endings
C:\IoT\EclipseKura\EclipseInstaller\iot-kura-develop-2017\git\kura\target-platform\com.codeminders.hidapi\src\main\java\com\codeminders\hidapi\HIDDevice.java:1: error: Do not use Windows line endings
C:\IoT\EclipseKura\EclipseInstaller\iot-kura-develop-2017\git\kura\target-platform\com.codeminders.hidapi\src\main\java\com\codeminders\hidapi\HIDDeviceInfo.java:1: error: Do not use Windows line endings
C:\IoT\EclipseKura\EclipseInstaller\iot-kura-develop-2017\git\kura\target-platform\com.codeminders.hidapi\src\main\java\com\codeminders\hidapi\HIDDeviceNotFoundException.java:1: error: Do not use Windows line endings
C:\IoT\EclipseKura\EclipseInstaller\iot-kura-develop-2017\git\kura\target-platform\com.codeminders.hidapi\src\main\java\com\codeminders\hidapi\HIDManager.java:1: error: Do not use Windows  line endings
C:\IoT\EclipseKura\EclipseInstaller\iot-kura-develop-2017\git\kura\target-platform\com.codeminders.hidapi\src\main\java\com\codeminders\hidapplet\HidApplet.java:1: error: Do not use Windows line endings
Audit done.
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] target-platform ................................... SUCCESS [1.095s]
[INFO] Camel :: Core :: Import sun.misc .................. SUCCESS [0.571s]
[INFO] System extension fragment for 'sun.misc' .......... SUCCESS [0.046s]
[INFO] Java API for working with Human Interface USB Devices (HID)  FAILURE [0.376s]
[INFO] usb4java .......................................... SKIPPED
[INFO] usb4java-javax .................................... SKIPPED
[INFO] Serial Device based on SODA DK comm ............... SKIPPED
[INFO] jdk.dio-parent .................................... SKIPPED
[INFO] jdk.dio ........................................... SKIPPED
[INFO] jdk.dio.armv5sf ................................... SKIPPED
[INFO] jdk.dio.armv6hf ................................... SKIPPED
[INFO] jdk.dio.x86 ....................................... SKIPPED
[INFO] jdk.dio.x86_64 .................................... SKIPPED
[INFO] gwt-user .......................................... SKIPPED
[INFO] moquette-broker ................................... SKIPPED
[INFO] p2-repo-common .................................... SKIPPED
[INFO] p2-repo-equinox_3.11.1 ............................ SKIPPED
[INFO] p2-repo-tests-deps ................................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------

--- a/README.md
+++ b/README.md
@@ -48,3 +48,14 @@ At first startup Eclipse IDE will checkout the code, perform a full build and co
 For further information, building and deployment instructions, please visit the [getting started page](https://wiki.eclipse.org/Kura/Getting_Started).
 
 To raise an issue, please report a bug on [GitHub issues](https://github.com/eclipse/kura/issues/new).
+
+## Supported Development Platforms
+The Eclipse Installer based setup works for the main used platforms like Linux, Mac Os and Windows.
+
+
+## Known issues
+Currently, the emulator web ui is not properly working on Windows so, with your setup, you will be able to build and deploy you applications, but not be able to use the Eclipse IDE based Kura emulator.
+
+The full build of Kura is only supported for Linux and Mac Os based systems.
+
+Currently the maven build on Windows requires to disable the tests and will fail when it tries to create the installers for the target platforms.

